### PR TITLE
Add `.editorconfig`

### PR DESCRIPTION
**Pull request by Pesa**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,20 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+
+[*.{cpp,py,ts}]
+indent_style = space
+indent_size = 4
```
Comment (Pesa) [inline .editorconfig:10]: 4 spaces are consistently used in the code snippets, so this should be uncontroversial.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+trim_trailing_whitespace = true
+
+[*.md]
+indent_style = space
+indent_size = 4
```
Comment (Pesa) [inline .editorconfig:15]: There are some inconsistencies in existing markdown files (a few bullet lists use 1 or 2 spaces) but the vast majority uses 4 spaces. We can settle on 2 or 4 spaces, up to you.

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+trim_trailing_whitespace = false
+
+[*.{yaml,yml}]
+indent_style = space
+indent_size = 2
```
Comment (Pesa) [inline .editorconfig:20]: This is the de facto standard for YAML files everywhere.

**Comment by pulsejet** [comment]
Nice

**Pull request merged**